### PR TITLE
add personas

### PR DESCRIPTION
**Pull request by ryanatkn**
This adds personas to the database and client session data and it adds an API route to create new ones. You should be able to run `gro db/create` which seeds some accounts and personas, and then from the homepage, after logging in you should see the account's personas and add new ones via a form.

The first three commits are from #40 and I'll rebase on master once that's merged.

### CODE REVIEW

File: src/db/migrations/20200403163837_init.ts [hunk 1]
```diff
@@ -15,4 +15,18 @@ export const up = async (knex: KnexInstance) => {
...
+
+	await knex.schema.createTable('personas', (t) => {
+		t.increments();
+		t.integer('account').unsigned().notNullable();
+		t.foreign('account').references('id').inTable('accounts');
+		// TODO does this need an index for searching duplicate names within each community?
+		// or can we encode this constraint in the database somehow?
+		// I tried to do `t.primary(['account', 'name'])` but got this error:
+		// 		alter table "personas" add constraint "personas_pkey" primary key ("account", "name") - multiple primary keys for table "personas" are not allowed
+		t.text('name').notNullable();
```
Comment (ryanatkn) [inline src/db/migrations/20200403163837_init.ts:27]: Does it make sense to use a composite primary key for this table, or is that weird? e.g. `t.primary(['account', 'name'])` instead of `t.increments();` to encode the uniqueness constraint? I think I'm overthinking it and the answer is no.

File: src/client/persona/CreatePersona.svelte [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<script>
```
Comment (ryanatkn) [inline src/client/persona/CreatePersona.svelte:1]: This file is just a bunch of copypasta from the login form. We'll revisit both.

File: src/accounts/AccountsRepo.test.ts [hunk 3]
```diff
@@ -77,6 +83,10 @@ test('AccountsRepo', () => {
 			t.ok(!result.ok);
 			t.is(result.type, 'invalidEmail');
 		});
+		test('errors with a null email', async () => {
+			// bypass the type system as a security check - accounts are extra important!
```
Comment (greatbacon) [inline src/accounts/AccountsRepo.test.ts:87]: Great corner case testing!

File: src/db/loadClientSession.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import {AccountSession} from '../client/session/session.js';
+import {unwrap} from '../utils.js';
+import {Db} from '../db/Db.js';
+import {AccountModel} from '../accounts/AccountModel.js';
+
+// TODO should this and `src/client/session` stuff be located in the same place?
+// should that be in `src/session`?
+// This is currently separated because it includes server-only imports.
+// Even though Rollup allows it, I think it's inadvisable
+// to mix server-only concerns into modules imported by the client.
```
Comment (greatbacon) [inline src/db/loadClientSession.ts:10]: I agree; we should try and keep server/client stuff as separated as possible

File: src/db/migrations/20200403163837_init.ts (hunk 1 above)
Comment (greatbacon) [reply src/db/migrations/20200403163837_init.ts:27]: I was gonna type up a big blob of a post but this might just be easier to chat over synchronously and then post a follow up comment on a decision. Let me know when you get this and we can schedule a quick chat.

File: src/personas/PersonasRepo.test.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+	test('create()', async () => {
+		test('creates a persona with a case-insensitive name', async () => {
+			const result = await personasRepo.create({account: testAccount, name: testNameUppercase});
+			t.ok(result.ok);
+			t.is(result.value.name, testName);
+			testPersona = result.value;
+		});
+		test('creates a second persona for the same account', async () => {
+			const name = randomName();
```
Comment (greatbacon) [inline src/personas/PersonasRepo.test.ts:26]: Would it make more sense to pull this out into the CONST block at the top of the test file?

Comment (ryanatkn) [reply src/personas/PersonasRepo.test.ts:26]: Good call, as written it's never used again, but later tests **should** use it to ensure querying works across both accounts. That was the point of having two accounts in the tests in the first place. I'll update the tests.

File: src/db/migrations/20200403163837_init.ts (hunk 1 above)
Comment (greatbacon) [reply src/db/migrations/20200403163837_init.ts:27]: Follow up from today's conversation: compound keys like this are usually a code smell in SQL. The incremental element as primary key is good enough for now, and any potential uniqueness concerns can be addressed via code for now.

**Review by greatbacon — APPROVED** [review]
Tested locally and made a few personas. LGTM!

**Comment by ryanatkn** [comment]
Most excellent.
